rust/private/rust.bzl

```diff
@@ -230,14 +233,14 @@ def _rust_test_common(ctx, toolchain, output):
     _assert_no_deprecated_attributes(ctx)
 
     crate_name = name_to_crate_name(ctx.label.name)
-
+    crate_type = "bin"
     if ctx.attr.crate:
         # Target is building the crate in `test` config
         # Build the test binary using the dependency's srcs.
         crate = ctx.attr.crate[rust_common.crate_info]
         target = rust_common.crate_info(
             name = crate_name,
-            type = crate.type,
+            type = crate_type,
```

Comment:
Should this assert that `crate.type == "bin"`?

Reply:
I don't think so? Crate under test can be a rust_library with crate type rlib or pretty much any other, it's only the test that will be "bin" in any case.